Parse/Internal/Persistence/Group/PFPersistenceGroup.h

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2015-present, Parse, LLC.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import <Parse/PFConstants.h>
+
+@class BFTask PF_GENERIC(id);
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol PFPersistenceGroup <NSObject>
+
+///--------------------------------------
+/// @name Data
+///--------------------------------------
+
+- (BFTask PF_GENERIC(NSData *)*)getDataAsyncForKey:(NSString *)key;
+
+- (BFTask *)setDataAsync:(NSData *)data forKey:(NSString *)key;
+- (BFTask *)removeDataAsyncForKey:(NSString *)key;
+
+- (BFTask *)removeAllDataAsync;
+
+///--------------------------------------
+/// @name Access
+///--------------------------------------
+
+- (void)beginLockedContentAccessToDataForKey:(NSString *)key;
```

Comment:
Could we replace this with a block-based or task-based API instead of requiring manual begin/end? Certainly would allow handling of error cases better.